test/IRGen/Inputs/usr/include/Gizmo.h

```diff
@@ -162,4 +162,6 @@ __attribute__((swift_name("OuterType.InnerType")))
 
 @protocol P
 - (oneway void)stuff;
+@property (nonatomic, readonly, getter=isValid) signed char valid;
+@property (nonatomic) NSString *name;
```

Comment:
Could we get some optional methods and properties in here, with a mix of class/instance for each? There are a couple of lists in the metadata here and I'd feel better if we could check all of them.

Also and/or alternatively, should the test for this code path be expressed in terms of an `@objc` protocol written in Swift? There's a case to be made that Clang should produce the metadata for an imported protocol directly, so there's no reimplementation and no way that the protocol metadata could get out of sync.

Reply:
Also, `@property(class)` defines a class property in Objective-C.

Reply:
I added class/instance required/optional methods in both the Objective-C protocol and a Swift defined Objective-C protocol.